packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/marker.dart

```diff
@@ -150,7 +150,12 @@ class Marker implements MapsObject<Marker> {
     this.position = const LatLng(0.0, 0.0),
     this.rotation = 0.0,
     this.visible = true,
+    @Deprecated(
+      'Use zIndexInt instead. '
+      'On iOS zIndex is truncated to an int, which can lead to incorrect/unstable ordering.',
+    )
     this.zIndex = 0.0,
+    this.zIndexInt = 0,
```

Comment:
zIndexInt should not be a new field, as this creates the possibility of inconsistent values in the same marker. Instead, it can be:

```dart
  const Marker({
    ...
    double zIndex = 0.0,
    int zIndexInt = 0,
    ...
  }): assert(zIndex == 0.0 || zIndexInt == 0, 'Only one of zIndex and zIndexInt can be provided'), zIndex = zIndexInt == 0 ? zIndex : zIndexInt.toDouble();

  ...
  /// ...
  // TODO(stuartmorgan): Make this an int when removing the deprecated double zIndex parameter.
  @Deprecated(...)
  final double zIndex;

  /// ...
  int get zIndexInt => zIndex.round();
```

Then there is only a single value. iOS can read `zIndexInt` since it will be truncated anyway, and Android can read `zIndex` with a comment explaining that it's using a deprecated parameter for backward compatibility.